src/utils/utils.js

```diff
@@ -1,88 +1,112 @@
-exports.isNode = function() {
-	return typeof process !== 'undefined' && process.toString() === '[object process]';
-};
+/**
+ * A regular expression that matches whitespace on either side, but
+ * not in the center of a string
+ *
+ * @type {RegExp}
+ */
+var TRIM_REGULAR_EXPRESSION = /^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g;
+
+/**
+ * Used in typeof comparisons
+ *
+ * @type {String}
+ */
+var OBJECT = 'object';
 
+/**
+ * True if environment is node, false if it's a browser
+ * This seems somewhat inelegant, if anyone knows a better solution,
+ * let's change this (must identify browserify's pseudo node implementation though)
+ *
+ * @public
+ * @type {Boolean}
+ */
+exports.isNode = typeof process !== 'undefined' && process.toString() === '[object process]';
+
+/**
+ * Provides as soon as possible async execution in a cross
+ * platform way
+ *
+ * @param   {Function} fn the function to be executed in an asynchronous fashion
+ *
+ * @public
+ * @returns {void}
+ */
 exports.nextTick = function( fn ) {
-	if( exports.isNode() ) {
+	if( exports.isNode ) {
 		process.nextTick( fn );
 	} else {
 		setTimeout( fn, 0 );
 	}
 };
 
+/**
+ * Removes whitespace from the beginning and end of a string
+ *
+ * @param   {String} inputString
+ *
+ * @public
+ * @returns {String} trimmedString
+ */
 exports.trim = function( inputString ) {
 	if( inputString.trim ) {
 		return inputString.trim();
 	} else {
-		return inputString.replace( trimRegExp, '' );
+		return inputString.replace( TRIM_REGULAR_EXPRESSION, '' );
```

Comment:
It's a polyfill for trim (see line 52/53) the only change is that I renamed the variable